Bug: Arrays as struct fields not supported

## Description
Struct fields cannot be declared as array types. Attempting to declare a struct with an array field type results in parser errors, preventing the creation of complex nested data structures.

## Reproduction

```ez
import @std

const Employee struct {
    name string
    age int
}

const Company struct {
    name string
    employees [Employee]  // This causes parser error
}

do main() {
    using std
    
    temp company Company = Company{
        name: "TechCorp",
        employees: {
            Employee{name: "Alice", age: 30},
            Employee{name: "Bob", age: 25}
        }
    }
    
    println("Company:", company.name)
    println("Employees:", len(company.employees))
}
```

## Current Output

```
error[E1002]: expected IDENT, got [ instead
  --> file.ez:10:15
   |
10 |     employees [Employee]
   |               ^ expected token not found
   |

error[E1001]: unexpected token '['
  --> file.ez:10:15
   |
10 |     employees [Employee]
   |               ^ unexpected token here
   |

error[E1003]: expected expression, found ']'
  --> file.ez:10:24
   |
10 |     employees [Employee]
   |                        ^ invalid syntax
   |
```

## Expected Behavior

Should be able to declare array-typed fields in structs, similar to other primitive and struct types:
- `fieldName [int]` - array of integers
- `fieldName [string]` - array of strings
- `fieldName [StructType]` - array of structs
- `fieldName [int, 5]` - fixed-size array

## Current Workaround

Currently must store arrays as separate variables and only store counts or metadata in struct fields:

```ez
const Company struct {
    name string
    employeeCount int  // Can only store count, not actual array
}
```

## Root Cause

The parser's struct field parsing does not support array type syntax. When parsing struct fields (likely in `pkg/parser/parser.go` around struct declaration parsing), it expects:
- Field name (IDENT)
- Field type (IDENT)

But array types start with `[` which is not recognized as a valid type token in this context.

## Impact

- **Severity**: MEDIUM - Language feature limitation
- **Frequency**: Affects any attempt to model complex nested data
- **User Experience**: Significantly limits expressiveness and requires awkward workarounds

## Use Cases Blocked

Without array-typed struct fields, users cannot:
1. Model one-to-many relationships naturally (Company has many Employees)
2. Create nested collections (Department has Teams, each Team has Members)
3. Store dynamic lists of data within structures
4. Implement common data structure patterns (Tree with children array, Graph with edges array)

## Related

This may require:
- Parser changes to recognize array types in struct field declarations
- Type system changes to validate array field types
- Evaluator changes to handle array field initialization and access

This is likely a significant feature addition rather than a simple bug fix, as it affects:
- Lexing/parsing of struct declarations
- AST representation of struct fields
- Type checking and validation
- Runtime struct initialization
- Memory management for nested structures

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -678,46 +678,8 @@ func (p *Parser) parseVarableDeclaration() *VariableDeclaration {
 
 	// Parse type - can be IDENT or [type] for arrays
 	p.nextToken()
-	if p.currentTokenMatches(LBRACKET) {
-		// Array type [type] or [type, size]
-		typeName := "["
-		p.nextToken() // move past [
-
-		// The element type should be an identifier
-		if !p.currentTokenMatches(IDENT) {
-			msg := fmt.Sprintf("expected type name, got %s", p.currentToken.Type)
-			p.errors = append(p.errors, msg)
-			p.addEZError(errors.E1007, msg, p.currentToken)
-			return nil
-		}
-		typeName += p.currentToken.Literal
-
-		// Check for fixed-size array syntax: [type, size]
-		if p.peekTokenMatches(COMMA) {
-			p.nextToken() // consume comma
-			p.nextToken() // get size
-
-			// Size should be an integer
-			if !p.currentTokenMatches(INT) {
-				msg := fmt.Sprintf("expected integer for array size, got %s", p.currentToken.Type)
-				p.errors = append(p.errors, msg)
-				p.addEZError(errors.E1007, msg, p.currentToken)
-				return nil
-			}
-			typeName += "," + p.currentToken.Literal
-		}
-
-		if !p.expectPeek(RBRACKET) {
-			return nil
-		}
-		typeName += "]"
-		stmt.TypeName = typeName
-	} else if p.currentTokenMatches(IDENT) {
-		stmt.TypeName = p.currentToken.Literal
-	} else {
-		msg := fmt.Sprintf("expected type, got %s", p.currentToken.Type)
-		p.errors = append(p.errors, msg)
-		p.addEZError(errors.E1007, msg, p.currentToken)
+	stmt.TypeName = p.parseTypeName()
+	if stmt.TypeName == "" {
 		return nil
 	}
 
@@ -1374,6 +1336,55 @@ func (p *Parser) parseEnumDeclaration() *EnumDeclaration {
 	return stmt
 }
 
+// parseTypeName parses a type name which can be:
+// - Simple type: int, string, StructName
+// - Array type: [int], [string], [StructName]
+// - Fixed-size array: [int,5], [string,10]
+func (p *Parser) parseTypeName() string {
+	if p.currentTokenMatches(LBRACKET) {
+		// Array type [type] or [type, size]
+		typeName := "["
+		p.nextToken() // move past [
+
+		// The element type should be an identifier
+		if !p.currentTokenMatches(IDENT) {
+			msg := fmt.Sprintf("expected type name, got %s", p.currentToken.Type)
+			p.errors = append(p.errors, msg)
+			p.addEZError(errors.E1007, msg, p.currentToken)
+			return ""
+		}
+		typeName += p.currentToken.Literal
+
+		// Check for fixed-size array syntax: [type, size]
+		if p.peekTokenMatches(COMMA) {
+			p.nextToken() // consume comma
+			p.nextToken() // get size
+
+			// Size should be an integer
+			if !p.currentTokenMatches(INT) {
+				msg := fmt.Sprintf("expected integer for array size, got %s", p.currentToken.Type)
+				p.errors = append(p.errors, msg)
+				p.addEZError(errors.E1007, msg, p.currentToken)
+				return ""
+			}
+			typeName += "," + p.currentToken.Literal
+		}
+
+		if !p.expectPeek(RBRACKET) {
+			return ""
+		}
+		typeName += "]"
+		return typeName
+	} else if p.currentTokenMatches(IDENT) {
+		return p.currentToken.Literal
+	} else {
+		msg := fmt.Sprintf("expected type, got %s", p.currentToken.Type)
+		p.errors = append(p.errors, msg)
+		p.addEZError(errors.E1007, msg, p.currentToken)
+		return ""
+	}
+}
+
 func (p *Parser) parseStructDeclaration() *StructDeclaration {
 	stmt := &StructDeclaration{Token: p.currentToken}
 
@@ -1406,10 +1417,14 @@ func (p *Parser) parseStructDeclaration() *StructDeclaration {
 			fieldNames[field.Name.Value] = field.Name.Token
 		}
 
-		if !p.expectPeek(IDENT) {
+		// Move to type token (can be IDENT or LBRACKET for arrays)
+		p.nextToken()
+
+		// Parse the type name (handles both simple types and array types)
+		field.TypeName = p.parseTypeName()
+		if field.TypeName == "" {
 			return nil
 		}
-		field.TypeName = p.currentToken.Literal
 
 		stmt.Fields = append(stmt.Fields, field)
 		p.nextToken()

